scripts/west_commands/sign.py

```diff
@@ -350,12 +350,22 @@ def edt_flash_params(flash):
         if align == 0:
             log.die('expected nonzero flash alignment, but got '
                     'DT flash device write-block-size {}'.format(align))
-        reg = image_0.regs[0]
-        if reg.size == 0:
-            log.die('expected nonzero slot size, but got '
-                    'DT image-0 partition size {}'.format(reg.size))
 
-        return (align, reg.addr, reg.size)
+        # The partitions node, and its subnode, must provide
+        # the size of image-1 or image-0 partition via the regs property.
+        image_key = 'image-1' if 'image-1' in images else 'image-0'
+        if not images[image_key].regs:
+            log.die('DT zephyr,flash node has no regs property;',
+                    "can't determine size of {}".format(image_key))
```

Comment:
Since this is an error message that users will see, I think this should be changed to refer to the exact node which is missing `regs`. Right now, it's referring to a different node, which might be confusing for people.

The chosen `zephyr,flash` node is something like `/soc/flash-controller@foo/flash@0`. That node is not missing a `regs` property.

Its grandchild node referring to a partition (something like `/soc/flash-controller@foo/flash@0/partitions/partition@bar`) is actually what is missing a `regs`.

```suggestion
            log.die(f'{image_key} flash partition has no regs property;',
                    "can't determine size of image")
```

Reply:
Updated as suggested.